module-4/backend/week_four.md

```diff
@@ -1,19 +1,57 @@
 ## Week Four - Module 4 Recap
 
-Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week. 
+1. What's your favorite project management tool?
 
-Note: When you're done, submit a PR. 
+Waffle.io
 
-1. What's your favorite project management tool?
 2. Why do we create staging environments?
+
+To ensure deployment is pain free and compatible across many machines. 
+
 3. What are the characteristics of a good README (in your opinion)?
+
+- Clear requirements for versions
+- Clear setup instructions
+- Maybe some screen shots of the in progress project
+- A description of the app or repo
+- Possible contribution instructions if it's open source
+
 4. What's one main improvement you're going to make to your code regarding accessibility issues?
+
+Better READMEs on all my repo's. Reactive styling so that content can be viewed on any device. Adding ALT's to all images for screen readers. 
+
 5. What are some basic security concerns to be aware of when building applications?
+
+- Broken Authentications
+- Injection
+- Insufficient logging and monitoring.
+
+
 6. Why is continuous integration helpful/important?
+
+Allows for feedback while in development which generally leads to a better product that fits the customers needs more closely. 
+
 7. What are some continuous integration tools?
 
+Unsure of this one. I would say that Trello or some other service that allows people to create issues or give feedback would be a solid approach. 
+
 #### Review  
 
 8. What are some characteristics of "good" git workflow?
+
+- Clear commit messages
+- Branches for grouped features
+- Branhcing off development
+- Merging locally before making a PR
+- Getting all PR's approved before merging
+
 9. What are the four fundamental concepts of object oriented programming?
+
+- Abstraction
+- Polymorphism
+- Inheritance
+- Encapsulation
+
 10. What's a module in Ruby and what's the difference between a class and a module?
+
+A module has shared functionality that can be passed between classes. You cannot instantiate a module. Those are the differences that come to mind with my current understanding. 
```

Comment:
those differences are correct, but i'd love for you to expand on what's the purpose of a module in Ruby? why do modules exist? what are some use cases?